### ISSUE TITLE
undefined method `chop' for nil:NilClass

### ISSUE THREAD

**Issue by leegarrett** [meringu/code_deploy#10]
Occasionally I get following error during deploy:

```
Recipe: code_deploy::default
  * code_deploy_agent[codedeploy-agent] action install

    ================================================================================
    Error executing action `install` on resource 'code_deploy_agent[codedeploy-agent]'
    ================================================================================

    NoMethodError
    -------------
    undefined method `chop' for nil:NilClass

    Cookbook Trace:
    ---------------
    /var/cache/chef/cookbooks/code_deploy/libraries/helper.rb:12:in `code_deploy_bucket'
    /var/cache/chef/cookbooks/code_deploy/libraries/helper.rb:26:in `https_code_deploy_bucket_loc'
    /var/cache/chef/cookbooks/code_deploy/libraries/helper.rb:58:in `version_hash'
    /var/cache/chef/cookbooks/code_deploy/libraries/helper.rb:78:in `installer_cache_path'
    /var/cache/chef/cookbooks/code_deploy/libraries/helper.rb:35:in `download_installer_to_cache'
    /var/cache/chef/cookbooks/code_deploy/resources/agent_debian.rb:21:in `block in class_from_file'

    Resource Declaration:
    ---------------------
    # In /var/cache/chef/cookbooks/code_deploy/recipes/default.rb

      1: code_deploy_agent 'codedeploy-agent'

    Compiled Resource:
    ------------------
    # Declared in /var/cache/chef/cookbooks/code_deploy/recipes/default.rb:1:in `from_file'

    code_deploy_agent("codedeploy-agent") do
      action [:install, :enable, :start]
      default_guard_interpreter :default
      declared_type :code_deploy_agent
      cookbook_name "code_deploy"
      recipe_name "default"
    end

    System Info:
    ------------
    chef_version=14.6.47
    platform=ubuntu
    platform_version=16.04
    ruby=ruby 2.5.3p105 (2018-10-18 revision 65156) [x86_64-linux]
    program_name=/usr/bin/chef-client
    executable=/opt/chef/bin/chef-client


```

The relevant snippet from helper.rb is:
```
    def code_deploy_bucket
      region = if node.key?('ec2')
                 node['ec2']['placement_availability_zone'].chop
               else
                 'us-east-1'
               end
      "aws-codedeploy-#{region}"
    end
```

The problem here is that just because `node['ec2']` exist, it doesn't mean that `node['ec2']['placement_availability_zone']` does. It seems as though the key is also named different ("availability_zone"), [looking up the ec2 plugin in ohai](https://github.com/chef/ohai/blob/master/lib/ohai/plugins/ec2.rb#L131).

**Comment by leegarrett** [comment]
Fixed with https://github.com/meringu/code_deploy/pull/11

**Comment by meringu** [comment]
released with 1.0.2

**Comment by leegarrett** [comment]
Much appreciated, thanks!
